src/notifications/common.ts

```diff
@@ -283,11 +285,17 @@ export const generateUserNotificationUniqueKey = ({
   return [uniqueKey, referenceId, referenceType].filter(Boolean).join(':');
 };
 
-export const postStatusToTypeMap: Record<
-  SourcePostModerationStatus,
-  NotificationType
-> = {
-  [SourcePostModerationStatus.Approved]: NotificationType.SourcePostApproved,
-  [SourcePostModerationStatus.Rejected]: NotificationType.SourcePostRejected,
-  [SourcePostModerationStatus.Pending]: NotificationType.SourcePostSubmitted,
+export const cleanupSourcePostModerationNotifications = async (
+  con: ConnectionManager,
+  post: ChangeObject<SourcePostModeration>,
+) => {
+  if (!post?.id) {
+    return;
+  }
+
+  await con.getRepository(NotificationV2).delete({
```

Comment:
I am not sure we want to delete notification, they are usually immutable and can reference other stuff, maybe introduce soft delete cc: @idoshamun ?

Reply:
let me search for some references. I know for sure that when we delete a post we also hide the notifications. Not sure if we delete or soft delete

Reply:
So I guess we do delete notifications 😂

Reply:
:triggered: